backend/src/middleware/validUrlMiddleware.js

```diff
@@ -0,0 +1,18 @@
+const validURL = str => {
+  const isValid = str.match(/^(?:https?:\/\/)(?:[^@\n])?(?:www\.)?([^:/\n?]+)/g)
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-04-12T15:37:10Z" title="Friday, April 12th 2019, 5:37:10 pm +02:00">Apr 12, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,18 @@
+const validURL = str => {
+  const isValid = str.match(/^(?:https?:\/\/)(?:[^@\n])?(?:www\.)?([^:/\n?]+)/g)
```


Would be great to see this unit tested. No need for the database to be involved.

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-04-12T15:38:03Z" title="Friday, April 12th 2019, 5:38:03 pm +02:00">Apr 12, 2019</time>_

---


E.g. I wonder if https is required or not.

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-04-12T15:38:25Z" title="Friday, April 12th 2019, 5:38:25 pm +02:00">Apr 12, 2019</time>_

---


and what about a missing domain? Like: `www.google`

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-04-12T16:08:59Z" title="Friday, April 12th 2019, 6:08:59 pm +02:00">Apr 12, 2019</time>_

---


I am not an expert on regex, but I believe I tested this and the `s` is optional, which is why it has a `?` on the right of it?

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-04-12T16:12:00Z" title="Friday, April 12th 2019, 6:12:00 pm +02:00">Apr 12, 2019</time>_

---


not sure what you mean by missing domain... as in no `.com`, etc?
I think this is the regex you came up with and I copied from `my-social-media.vue`... could be that we could improve on it...

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-04-12T16:15:35Z" title="Friday, April 12th 2019, 6:15:35 pm +02:00">Apr 12, 2019</time>_

---


I'm sorry, I meant **top-level**-domain.

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-04-12T16:16:34Z" title="Friday, April 12th 2019, 6:16:34 pm +02:00">Apr 12, 2019</time>_

---


Up to you! If you want to get the PR merged, maybe create `good-first-issue` for refactoring?

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-04-12T16:29:53Z" title="Friday, April 12th 2019, 6:29:53 pm +02:00">Apr 12, 2019</time>_

---


can be, that sounds like a good compromise that would potentially allow more conversation on the subject